Lib/base64.py

```diff
@@ -198,8 +199,8 @@ def b32decode(s, casefold=False, map01=None):
     global _b32rev
     # Delay the initialization of the table to not waste memory
     # if the function is never called
-    if _b32rev is None:
-        _b32rev = {v: k for k, v in enumerate(_b32alphabet)}
+    if alphabet not in _b32rev:
```

Comment:
It feels mildly weird to me to hash the entire alphabet on every call, but I tried a different approach that uses sentinels (enums would also work) and found that it did not make an appreciable difference in speed, and this *is* more elegant — it has the nice property that you don't even need to change the code if additional base32 alphabets are added in the future — so I am in favor of keeping it like this.

Just thought I'd make a comment to explain my reasoning to future software archaeologists.

Reply:
> it has the nice property that you don't even need to change the code if additional base32 alphabets are added in the future

That was my goal :P